src/client/App.js

```diff
@@ -1,26 +1,57 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
 
+import * as itemsActions from './components/ProductsList/actions';
 import Header from './components/Header';
-import SearchBar from './components/SearchBar';
+import { SearchBar } from './components/SearchBar/index';
 import ProductsContainer from './components/ProductsContainer';
 import Footer from './components/Footer';
-import initialData from './common/initialData.json';
 
 class App extends Component {
-  state = initialData;
+  componentDidMount() {
+    this.fetchItems();
+  }
+
+  fetchItems = () => {
+    const {
+      itemsActions: { fetchItems }
+    } = this.props;
+    fetchItems();
+  };
 
   render() {
-    const { shoopingList, archiveList } = this.state;
+    const { archiveList, shoppingList } = this.props;
     return (
       <div className="app-wrapper">
         <Header />
         <SearchBar />
-        <ProductsContainer products={shoopingList} />
+        <ProductsContainer products={shoppingList} />
         <ProductsContainer isArchive products={archiveList} />
         <Footer />
       </div>
     );
   }
 }
 
-export default App;
+App.propTypes = {
+  archiveList: PropTypes.arrayOf(PropTypes.object),
+  itemsActions: PropTypes.shape({ fetchItems: PropTypes.func }),
+  shoppingList: PropTypes.arrayOf(PropTypes.object)
+};
+
+const mapStateToProps = state => ({
+  addItemSuccess: state.addItemSuccess,
+  archiveList: state.items.archiveList,
+  shoppingList: state.items.shoppingList
+});
+
+const mapDispatchToProps = dispatch => ({
+  itemsActions: bindActionCreators(itemsActions, dispatch)
```

Comment:
please import just one needed function

Reply:
This is fixed in the next pull request. The whole structure of app components is changed to /App directory with actions etc.